app.rb

```diff
@@ -0,0 +1,138 @@
+require './book'
+require './classroom'
+require './person'
+require './teacher'
+require './student'
+require './rental'
+
+class App
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  def console_entry_point
+    puts 'Welcome to my School Library'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thanks for using my School Library App'
+        break
+      end
+      option input
+    end
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'No Available books! Add a book'
+    else
+      @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+    end
+    back_to_menu
+  end
+
+  def list_all_persons
+    puts 'No person Available.' if @persons.empty?
+    @persons.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}"
+    end
+    back_to_menu
+  end
+
+  def create_person
+    print 'To create a student, press 1, to create a teacher, press 2 : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again'
+    end
+  end
+
+  def create_student
+    puts 'Create a new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      Student.new('classroom', age, name, parent_permission: false)
+      @persons << student
+      puts 'Student does not have parent permission, cant rent books'
+    when 'y'
+      student = Student.new('classroom', age, name, parent_permission: true)
+      @persons << student
+      puts 'Entry successful'
+    end
+  end
```

Comment:
- [ ] Please I am getting this error(see screenshot below) when I try to create a `student` or a `teacher`, please checkout the error message and fix the issue. It has to do with the number of parameters passed in to the initialize method in the `person class`. 
![error-message](https://user-images.githubusercontent.com/62849872/163402237-2e078df4-ba5c-4f02-a0c1-8a73a7efe5e3.png)